hypnotoad/cases/tokamak.py

```diff
@@ -457,6 +459,14 @@ def __init__(
             self.o_point = Point2D(opoints[0][0], opoints[0][1])
         self.psi_axis = psi_axis
 
+        if len(xpoints) == 0:
+            warnings.warn("No X-points found in TokamakEquilibrium input")
+        else:
+            if psi_bdry is None:
+                psi_bdry = xpoints[0][2]  # Psi on primary X-point
+            self.x_point = Point2D(xpoints[0][0], xpoints[0][1])
+        self.psi_bdry = psi_bdry
```

Comment:
I don't think we ever actually pass the `psi_axis` argument at the moment (and we won't be passing `psi_bndry` with this PR). I think it would be nicer to redefine those arguments to `__init__()` so we use them to pass in the values from the EFIT file. We can then store those values as member variables here so we don't have to read the gfile again when writing output (not that that's really a problem, I just think it's neater to only have to open it once). I think we should always use the 'found' psi value. It's probably also best to rename the argument `psi_bdry`->`psi_bdry_gfile`. It would be nice to add a sanity check comparing the found and EFIT values too. So something like:
```suggestion
            self.psi_bdry = xpoints[0][2]  # Psi on primary X-point
            self.x_point = Point2D(xpoints[0][0], xpoints[0][1])
            self.psi_bdry_gfile = psi_bdry_gfile
            if psi_bdry_gfile is not None and abs(
                self.psi_bdry - psi_bdry_gfile
            ) > 1.e-3:
                raise ValueError(
                    f"psi_bdry from the gfile ({psi_bdry}) is different from psi "
                    f"calculated at the primary X-point ({self.psi_bdry})"
                )
```
and something similar for `psi_axis`